src/node-uptimes/node-uptimes-loader.ts

```diff
@@ -16,26 +22,28 @@ export class NodeUptimesLoader {
     private readonly prisma: PrismaService,
   ) {}
 
-  async createEvent(user: User, occurredAt: Date): Promise<void> {
-    const uptime = await this.nodeUptimesService.get(user);
-    if (!uptime || uptime.total_hours < NODE_UPTIME_CREDIT_HOURS) {
-      return;
-    }
-
-    await this.prisma.$transaction(async (prisma) => {
-      const event = await this.eventsService.createNodeUptimeEventWithClient(
-        user,
-        occurredAt,
-        prisma,
-      );
-      if (!event) {
-        throw new Error(`Error creating node uptime event`);
+  async createEvents(props: CreateEventsProps): Promise<void> {
+    for (const { user, occurredAt } of props) {
+      const uptime = await this.nodeUptimesService.get(user);
+      if (!uptime || uptime.total_hours < NODE_UPTIME_CREDIT_HOURS) {
+        return;
       }
 
-      await this.nodeUptimesService.decrementCountedHoursWithClient(
-        uptime,
-        prisma,
-      );
-    });
+      await this.prisma.$transaction(async (prisma) => {
```

Comment:
We could optimize these functions to do bulk update, but I left this way now to make sure direction seems sound.